org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/YAMLLanguageServer.java

```diff
@@ -53,27 +87,40 @@ public YAMLLanguageServer() {
 	
 	@Override
 	public void handleMessage(Message message, LanguageServer languageServer, URI rootUri) {
+		if (message instanceof ResponseMessage) {
+			ResponseMessage responseMessage = (ResponseMessage) message;
+			if (responseMessage.getResult() instanceof InitializeResult) {
+				Map<String, Object> settings = new HashMap<>();
+				settings.put(YAML_KEY, getYamlSchemaOptions());
+				
+				DidChangeConfigurationParams params = new DidChangeConfigurationParams(settings);
+				languageServer.getWorkspaceService().didChangeConfiguration(params);
+			}
+		}
+	}
+
+	private static Map<String, Object> getYamlSchemaOptions() {
+		Map<String, Object> yaml = new HashMap<>();
 		IPreferenceStore preferenceStore = Activator.getDefault().getPreferenceStore();
 		String schemaStr = preferenceStore.getString(YAMLPreferenceInitializer.YAML_SCHEMA_PREFERENCE);
-		if (cachedSchema == null || !schemaStr.equals(cachedSchema)) {
-			cachedSchema = schemaStr;
+		if (!schemaStr.trim().isEmpty()) {
 			Map<String, Object> schemas = new Gson().fromJson(schemaStr, new TypeToken<HashMap<String, Object>>() {}.getType());
-			Map<String, Object> yaml = new HashMap<>();
-			yaml.put("schemas", schemas);
-			yaml.put("validate", true);
-			yaml.put("completion", true);
-			yaml.put("hover", true);
-			
-			Map<String, Object> settings = new HashMap<>();
-			settings.put("yaml", yaml);
-			
-			DidChangeConfigurationParams params = new DidChangeConfigurationParams(settings);
-			languageServer.getWorkspaceService().didChangeConfiguration(params);
+			yaml.put(SCHEMAS_KEY, schemas);
+			yaml.put(VALIDATE_KEY, true);
+			yaml.put(COMPLETION_KEY, true);
+			yaml.put(HOVER_KEY, true);
 		}
+		return yaml;
 	}
-	
+
 	@Override
 	public String toString() {
 		return "YAML Language Server: " + super.toString();
 	}
+
+	@Override
+	protected void finalize() throws Throwable {
```

Comment:
finalize is deprecated and shouldn't be used at all. There is a `stop` method you can override instead.

Reply:
Ah. Sure. Thanks for the point